src/components/notification-picker.tsx

```diff
@@ -0,0 +1,226 @@
+    setSendToInput(e.target.value);
+  }, []);
+
+  const blur = useCallback(() => {
+    const emailArray = sendToInput
+      .split(',')
+      .map(email => email.trim())
+      .filter(email => email);
+    const updatedNotification = { ...model.notification, sendTo: emailArray };
+    modelChange({ ...model, notification: updatedNotification });
+  }, [sendToInput, model, modelChange]);
+
+  const keyDown = useCallback(
+    (e: React.KeyboardEvent) => {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        blur();
+      }
+    },
+    [blur]
+  );
+
+  const includeOutputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const updatedValue = event.target.checked;
+    setIncludeOutput(updatedValue);
+    modelChange({
+      ...model,
+      notification: {
+        ...model.notification,
+        includeOutput: updatedValue
+      }
+    });
+  };
+
+  const deleteSelectedEvent = useCallback(
+    (eventToDelete: string) => () => {
+      const updatedEvents = selectedEvents.filter(
+        event => event !== eventToDelete
+      );
+      setSelectedEvents(updatedEvents);
+      modelChange({
+        ...model,
+        notifications: { ...model.notification, selectedEvents: updatedEvents }
+      });
+    },
+    [selectedEvents, model, modelChange]
+  );
+
+  if (!notificationEvents.length) {
+    return null;
+  }
+
+  return (
+    <Stack size={2}>
+      <InputLabel>{trans.__('Notification')}</InputLabel>
+      <FormControlLabel
+        control={
+          <Switch
+            checked={enableNotification}
+            onChange={enableNotificationChange}
+          />
+        }
+        label={trans.__('Enable Notification')}
+      />
+      <TextField
+        label={trans.__('Send To')}
+        value={sendToInput}
+        name="sendTo"
+        variant="outlined"
+        onChange={sendToChange}
+        onBlur={blur}
+        onKeyDown={keyDown}
+        disabled={!enableNotification}
+      />
+      <NotificationEventsSelect
+        id={id}
+        availableEvents={notificationEvents.filter(
+          e => !selectedEvents.includes(e)
+        )}
+        selectChange={selectChange}
+        disabled={!enableNotification}
+      />
+      <SelectedEventsChips
+        selectedEvents={selectedEvents}
+        deleteSelectedEvent={deleteSelectedEvent}
+        disabled={!enableNotification}
+      />
+      <FormControlLabel
+        control={
+          <Checkbox
+            checked={includeOutput}
+            onChange={includeOutputChange}
+            disabled={!enableNotification}
+          />
+        }
+        label={trans.__('Include Output')}
+      />
+    </Stack>
+  );
+}
+
+const NotificationEventsSelect: React.FC<{
+  id: string;
+  availableEvents: string[];
+  selectChange: (e: SelectChangeEvent<string>) => void;
+  disabled: boolean;
+}> = ({ id, availableEvents, selectChange, disabled }) => {
```

Comment:
1. Why define these components via arrow functions instead of plain old functions?  Compare the two in readability:

```ts
const Component: React.FC<ComponentProps> = (props) => { /* ... */ }

// v.s.

function Component(props: ComponentProps) { /* ... */ }
```

It's also worth noting that functions are [hoisted](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting) via const arrow functions are not. This is a pretty useful property in general, and is another reason why I prefer plain old functions.

2. Please extract the props type declaration into a separate type rather than doing it inline. Complex inline types are difficult to parse, and the lack of a name makes their purpose obscure until studied carefully.
3. nit: I see that you're using ES6 destructuring to avoid having to prefix `props.*` to every prop. While I totally understand the motivation to type less, that prefix is a helpful indicator that "this expression is a React prop" and distinguishes it from local variables which are generally less important helpers.

Reply:
Same goes for `SelectedEventsChips`.

Reply:
1. Using arrow functions allows to destructure props within the function declaration making code more concise and readable (vs destructuring in a separate line; if we would like to avoid destructuring props in our codebase as you suggest, this is less of an advantage). Using `React.FC<TYPE>` also provides explicit and standardized function signature that does not rely on a convention and manual typing of `props` which IMO makes code easier to read and write. 

I don't have a hard preference here and think that both approaches have merit as well as pros and cons. It seems like you do so I'm ok to use functions instead. Interestingly, Netlify's React Typescript Cheatsheet agrees with you and expresses opinion that React.FC might be unnecessary: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/

2. I think this is a question of a right balance between verbosity and clarity. In your opinion, should props type declarations always be separate types/interfaces? 

3. I think this is also a question of balance between verbosity and clarity.

Don't have a hard preference on 2 or 3 as well so will update in line with your comments.

Reply:
Should we use ESLint `func-style` rule allowing only declaration-type functions and explicitly disallowing arrow functions to formalize this informal preference for plain old functions into a formal requirement? This way we won't have to discuss this in the future. I can do this when time allows.

https://eslint.org/docs/latest/rules/func-style#options

Reply:
Arrow functions allow you to define an inline function very concisely, so they still have their uses in change handlers for example. So I don't think it's necessary to add this ESLint rule.